ako-gateway-api/k8s/validator.go

```diff
@@ -261,84 +261,78 @@ func validateParentReference(key string, httpRoute *gatewayv1beta1.HTTPRoute, ht
 		ObservedGeneration(httpRoute.ObjectMeta.Generation)
 
 	//section name is optional
-	//TODO check
-	if httpRoute.Spec.ParentRefs[index].SectionName == nil {
-		// can't attach to any so update the httproute status
-		utils.AviLog.Errorf("key: %s, msg: Section Name in Parent Reference %s is empty, HTTPRoute object %s cannot be attached to a listener", key, name, httpRoute.Name)
-		err := fmt.Errorf("Listener not specified")
-		defaultCondition.
-			Message(err.Error()).
-			SetIn(&httpRouteStatus.Parents[index].Conditions)
-		return err
-	}
-
-	listenerName := *httpRoute.Spec.ParentRefs[index].SectionName
-	httpRouteStatus.Parents[index].ParentRef.SectionName = &listenerName
-	i := akogatewayapilib.FindListenerByName(string(listenerName), gateway.Spec.Listeners)
-	if i == -1 {
-		// listener is not present in gateway
-		utils.AviLog.Errorf("key: %s, msg: unable to find the listener from the Section Name %s in Parent Reference %s", key, name, listenerName)
-		err := fmt.Errorf("Invalid listener name provided")
-		defaultCondition.
-			Message(err.Error()).
-			SetIn(&httpRouteStatus.Parents[index].Conditions)
-		return err
-	}
-
-	// TODO: Don't attach to a invalid listener configuration
-	// check from store
-
-	hostInListener := gateway.Spec.Listeners[i].Hostname
-
-	if hostInListener == nil {
-		utils.AviLog.Errorf("key: %s, msg: no hostname found in parent", key)
-		err := fmt.Errorf("No hostname found in parent")
-		defaultCondition.
-			Message(err.Error()).
-			SetIn(&httpRouteStatus.Parents[index].Conditions)
-		return err
+	var listenersForRoute []gatewayv1beta1.Listener
+	if httpRoute.Spec.ParentRefs[index].SectionName != nil {
+		listenerName := *httpRoute.Spec.ParentRefs[index].SectionName
+		httpRouteStatus.Parents[index].ParentRef.SectionName = &listenerName
+		i := akogatewayapilib.FindListenerByName(string(listenerName), gateway.Spec.Listeners)
+		if i == -1 {
+			// listener is not present in gateway
+			utils.AviLog.Errorf("key: %s, msg: unable to find the listener from the Section Name %s in Parent Reference %s", key, name, listenerName)
+			err := fmt.Errorf("Invalid listener name provided")
+			defaultCondition.
+				Message(err.Error()).
+				SetIn(&httpRouteStatus.Parents[index].Conditions)
+			return err
+		}
+		listenersForRoute = append(listenersForRoute, gateway.Spec.Listeners[i])
+	} else {
+		listenersForRoute = append(listenersForRoute, gateway.Spec.Listeners...)
 	}
 
-	// replace the wildcard character with a regex
-	replacedHostname := strings.Replace(string(*hostInListener), "*", "([a-zA-Z0-9-]{1,})", 1)
+	var listenersMatchedToRoute []gatewayv1beta1.Listener
```

Comment:
@aaha97 please add some UTs to test and also check whether we need to modify some UTs.

Reply:
Please take care of this in the next PR. I merging this one.